### PR TITLE
Fix #1854: Modified aggregation to use impact layer extent 

### LINKED ISSUE #1854: Fix zero post processor in Earthquake Polygon + Population
When we ran Earthquake Polygon Hazard and Population Exposure using the test data, the report succesfully computed but the postprocessor returns zero results.

### PR DESCRIPTION
**Pull request by lucernae**
Modify aggregation to use impact layer extent if aggregation layer is not provided.

This fix is used to handle cases where the hazard vector layer contains polygon which is outside the optimal extent but must be included in the postprocessor calculations.

### CODE REVIEW

**Comment by akbargumbira** [comment]
can we ask @mbernasocchi if this is a wise decision?

File: safe/impact_statistics/aggregator.py [hunk 1]
```diff
@@ -171,6 +171,23 @@ def __init__(self, extent, aggregation_layer):
...
+        return self._extent
+
+    @extent.setter
+    def extent(self, value):
+        self._extent = value
+        # update layer extent to match impact layer if in aoi_mode
+        if self.aoi_mode:
+            try:
+                self.layer = self._extents_to_layer()
+                self.safe_layer = safe_read_layer(str(self.layer.source()))
```
Comment (akbargumbira) [inline safe/impact_statistics/aggregator.py]: Remove casting it to str. It will raise unicode error if the layer source's got some non ASCII

Comment (lucernae) [inline safe/impact_statistics/aggregator.py]: Then, I will take note that we should also change the same statement in _prepare_layer

**Comment by lucernae** [comment]
Perhaps I should wait for further discussion for this PR

**Comment by mbernasocchi** [comment]
All Looks good to me

**Pull request merged**